File: db_core/order_notes.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""Canonical 备注 identity for dine-in 加菜 / 退菜 / 退后重下 / 对调."""

from typing import Any

_PLATFORM_NOTES_PREFIX = "外卖平台:"


def canonical_order_notes(value: Any) -> str:
    """Identity key for an order-line 备注.

    Strip; missing/blank is empty. Strings starting with 外卖平台: are not 备注.
    """
    if value is None:
        return ""
    text = str(value).strip()
    if text.startswith(_PLATFORM_NOTES_PREFIX):
        return ""
    return text

File: kds/utils/orderNotes.js
/**
 * Canonical 备注 identity and visible copy.
 * Same rule as Python db_core.order_notes.canonical_order_notes:
 * strip; strings starting with 外卖平台: are empty (not 备注).
 */

const PLATFORM_NOTES_PREFIX = '外卖平台:'

/**
 * @param {unknown} value
 * @returns {string}
 */
export function canonicalOrderNotes(value) {
  if (value == null) return ''
  const text = String(value).trim()
  if (text.startsWith(PLATFORM_NOTES_PREFIX)) return ''
  return text
}

/**
 * 菜卡 / 待上笼组 identity: 菜名 + 归一备注.
 * @param {string} dishName
 * @param {unknown} notes
 * @returns {string}
 */
export function dishNotesIdentityKey(dishName, notes) {
  return `${String(dishName || '')}\0${canonicalOrderNotes(notes)}`
}
